graveler/graveler.go

```diff
@@ -1043,15 +1049,22 @@ func (g *Graveler) ResetPrefix(ctx context.Context, repositoryID RepositoryID, b
 // Example: consider the following tree: C1 -> C2 -> C3, with the branch pointing at C3.
 // To revert C2, we merge C1 into the branch, with C2 as the merge base.
 // That is, try to apply the diff from C2 to C1 on the tip of the branch.
-func (g *Graveler) Revert(ctx context.Context, repositoryID RepositoryID, branchID BranchID, ref Ref, committer string, message string, metadata Metadata) (CommitID, error) {
+// If the commit is a merge commit, 'parentNumber' is the parent number (1-based) relative to which the revert is done.
+func (g *Graveler) Revert(ctx context.Context, repositoryID RepositoryID, branchID BranchID, ref Ref, parentNumber int, commitParams CommitParams) (CommitID, error) {
 	commitID, err := g.branchLocker.MetadataUpdater(ctx, repositoryID, branchID, func() (interface{}, error) {
 		commitRecord, err := g.getCommitRecordFromRef(ctx, repositoryID, ref)
 		if err != nil {
 			return "", fmt.Errorf("get commit from ref %s: %w", ref, err)
 		}
-		if len(commitRecord.Parents) > 1 {
-			// TODO(johnnyaug) support reverting merge commits
-			return "", ErrRevertMergeCommit
+
+		if len(commitRecord.Parents) > 1 && parentNumber <= 0 {
+			return "", ErrRevertMergeNoParent
+		}
+		if parentNumber > 0 {
+			if parentNumber > len(commitRecord.Parents) { // parent number is 1-based
+				return "", fmt.Errorf("%w: parent %d", ErrRevertParentOutOfRange, parentNumber)
+			}
+			parentNumber--
 		}
```

Comment:
1. parameter verification can be done outside the lock
2. Think we can consider 0 as no value was passed and assume the default (which is 1)